keps/sig-scheduling/5465-improve-resourcequota-with-schedulegated-pods/README.md

```diff
@@ -0,0 +1,1736 @@
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+This KEP proposes changes to the Kubernetes `ResourceQuota` enforcement mechanism to improve alignment with established scheduling paradigms. Specifically, it introduces a more lenient admission strategy for pods requesting CPU and memory resources that would exceed the namespace-scoped `ResourceQuota`, while relying on the scheduler to determine actual feasibility. The proposal includes:
+
+* Allowing admission of such overcommitting pods if they are blocked by a `PodSchedulingReadiness` gate.
+* Excluding these gated pods from `ResourceQuota` usage calculation.
+* Enhancing the scheduler to respect `ResourceQuota` limits and leave pods in the Pending phase if scheduling would cause quota overcommit.
+
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+Current `ResourceQuota` enforcement is strict: if a pod's resource request exceeds the remaining quota, it is rejected during admission. This is inconsistent with other Kubernetes paradigms, such as node resource availability, where unschedulable pods are allowed to remain in a Pending state.
```

Comment:
Wondering how in-place pod resizing works with ResourceQuota today? Is it correctly handled?
That could be another thing that we have to consider not to make any conflicts.  ...if they don't handle the resource quota correctly today, we don't have to consider them in this KEP, but instead, I'd have to put a blocker on the in-place pod resizing graduation.

Reply:
Ah... just remembered this: https://github.com/kubernetes/enhancements/issues/5465#issuecomment-3158097293